src/ros_filter.cpp

```diff
@@ -2971,12 +2973,32 @@ namespace RobotLocalization
 
     // Walk back through the queue until we reach a filter state whose time stamp is less than or equal to the
     // requested time. Since every saved state after that time will be overwritten/corrected, we can pop from
-    // the queue.
+    // the queue. If the history is insufficiently short, we just take the oldest state we have.
+    FilterStatePtr lastHistoryState;
     while (!filterStateHistory_.empty() && filterStateHistory_.back()->lastMeasurementTime_ > time)
     {
+      lastHistoryState = filterStateHistory_.back();
       filterStateHistory_.pop_back();
     }
 
+    // If the state history is not empty at this point, it means that our history was large enough, and we
+    // should revert to the state at the back of the history deque.
+    bool retVal = false;
+    if (!filterStateHistory_.empty())
+    {
+      retVal = true;
```

Comment:
You are reverting to a previous time regardless of whether you are reverting to the desired previous time or not, but you are only returning "true" if you reverted to the desired time. I just want to verify that is the desired meaning of the return code.

Reply:
Correct. The semantics of the return type are a little weird, I'll grant you.